lib/internal/DoublyLinkedList.js

```diff
@@ -60,3 +63,26 @@ DLL.prototype.shift = function() {
 DLL.prototype.pop = function() {
     return this.tail && this.removeLink(this.tail);
 };
+
+DLL.prototype.toArray = function () {
+    var arr = Array(this.length);
+    var idx = 0;
+    var curr = this.head;
+    for(idx = 0; idx < this.length; idx++) {
+        arr[idx] = curr.data;
+        curr = curr.next;
+    }
+    return arr;
+}
+
+DLL.prototype.filter = function (testFn) {
```

Comment:
Actually after more thought, `remove` is a better name.  If someone is using an array implementation, they're likely doing their own array splicing somehow anyway.